src/main/java/org/myrobotlab/service/InMoov2.java

```diff
@@ -1486,11 +1626,84 @@ public String publishFlash(String flashName) {
     return flashName;
   }
 
-  public String publishHeartbeat() {
-    if (config.heartbeatFlash) {
-      invoke("publishFlash", "heartbeat");
+  /**
+   * FIXME - get rid of all functionality here - should all be controlled by
+   * behaviors
+   * 
+   * A heartbeat that continues to check status, and fire events to the FSM.
+   * Checks battery, flashes leds and processes all the configured checks in
+   * onHeartbeat at a regular interval
+   */
+  public Heartbeat publishHeartbeat() {
```

Comment:
this does break with convention a bit.. normally we'd have a publishHeartbeat(Heartbeat beat)   ... that simply returns the Heartbeat passed... the code below here, could be handled in the heart thread itself instead..  and just invoking publishHeartbeat with the beat that it created in that cycle...  just curious why the departure from that convention?